frontend/lib/src/components/shared/Dropdown/Selectbox.tsx

```diff
@@ -88,222 +77,205 @@ export function fuzzyFilterSelectOptions(
   ).reverse()
 }
 
-export class Selectbox extends React.PureComponent<Props, State> {
-  public state: State = {
-    isEmpty: false,
-    value: this.props.value,
-  }
+const Selectbox: React.FC<Props> = ({
+  disabled,
+  width,
+  value: propValue,
+  onChange,
+  options: propOptions,
+  label,
+  labelVisibility,
+  help,
+  placeholder,
+  clearable,
+}) => {
+  const theme: EmotionTheme = useTheme()
+  const [isEmpty, setIsEmpty] = useState(false)
+  const [value, setValue] = useState<number | null>(propValue)
+
+  // Update the value whenever the value provided by the props changes
+  useEffect(() => {
+    setValue(propValue)
+  }, [propValue])
```

Comment:
**note:** For the future, it would be great to find an alternative to this approach since this is an unnecessary `useEffect` call, which causes re-renders [[React docs](https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state)]. It's fine for this PR since this is a 1->1 conversion, but perhaps an inline comment might be helpful for future reference

Reply:
Ok, I added a comment 👍